tempesta_fw/server.h

```diff
@@ -255,6 +262,33 @@ tfw_srv_conn_need_resched(TfwSrvConn *srv_conn)
 	return ((ACCESS_ONCE(srv_conn->recns) >= sg->max_recns));
 }
 
+/*
+ * Put server into alive or suspended (exluded from processing) state.
+ */
+static inline void
+tfw_srv_mark_alive(TfwServer *srv)
+{
+	if (test_bit(TFW_SRV_B_HMONITOR, (unsigned long *)&srv->hm_flags))
+		clear_bit(TFW_SRV_B_SUSPEND, (unsigned long *)&srv->hm_flags);
+}
```

Comment:
Thank you for helpful comment, `test_bit(TFW_SRV_B_HMONITOR, (unsigned long *)&srv->hm_flags)` is not needed here at all. I left this check in `tfw_srv_suspended()` to avoid spinlock (or read/writelock) synchronization in `tfw_http_hm_control()`.